numpy/core/src/umath/loops.c.src

```diff
@@ -1644,6 +1644,41 @@ TIMEDELTA_mm_q_floor_divide(char **args, npy_intp *dimensions, npy_intp *steps,
     }
 }
 
+NPY_NO_EXPORT void
+TIMEDELTA_mm_qm_divmod(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUSED(func))
+{
+    BINARY_LOOP_TWO_OUT {
+        const npy_timedelta in1 = *(npy_timedelta *)ip1;
+        const npy_timedelta in2 = *(npy_timedelta *)ip2;
+        if (in1 == NPY_DATETIME_NAT || in2 == NPY_DATETIME_NAT) {
+            npy_set_floatstatus_invalid();
+            *((npy_timedelta *)op1) = 0;
+            *((npy_timedelta *)op2) = NPY_DATETIME_NAT;
+        }
+        else if (in2 == 0) {
+            npy_set_floatstatus_divbyzero();
+            *((npy_timedelta *)op1) = 0;
+            *((npy_timedelta *)op2) = 0;
```

Comment:
Based on how `divmod(np.array(1.0), np.array(0))` behaves, this should be `NAT`.

Reply:
I'm not sure I understand -- I haven't added any new code here -- it just the `m8` remainder and floordiv code paths pasted into a new inner loop together.

If we do this, the modulus operator would produce a different result when used in isolation vs. in the context of divmod.

Reply:
Hmm, I didn't check the modulus code. You're right that what you have here is consistent with modulus.

On the other hand, I think we might have gotten modulus wrong. If we're going to change modulus on timedeltas, it might be better to do so before introducing divmod, so that we change behavior in one place and not two,